src/tigerbeetle/cli.zig

```diff
@@ -133,330 +164,152 @@ pub const Command = union(enum) {
-            replica_count = parse_flag("--replica-count", arg);
-        } else if (mem.startsWith(u8, arg, "--replica")) {
-            if (command != .format) fatal("--replica: supported only by 'format' command", .{});
-            replica = parse_flag("--replica", arg);
-        } else if (mem.startsWith(u8, arg, "--addresses")) {
-            if (command != .start) fatal("--addresses: supported only by 'start' command", .{});
-            addresses = parse_flag("--addresses", arg);
-        } else if (mem.startsWith(u8, arg, "--cache-accounts")) {
-            if (command != .start) fatal("--cache-accounts: supported only by 'start' command", .{});
-            cache_accounts = parse_flag("--cache-accounts", arg);
-        } else if (mem.startsWith(u8, arg, "--cache-transfers")) {
-            if (command != .start) fatal("--cache-transfers: supported only by 'start' command", .{});
-            cache_transfers = parse_flag("--cache-transfers", arg);
-        } else if (mem.startsWith(u8, arg, "--cache-transfers-posted")) {
-            if (command != .start) fatal("--cache-transfers-posted: supported only by 'start' command", .{});
-            cache_transfers_posted = parse_flag("--cache-transfers-posted", arg);
-        } else if (mem.startsWith(u8, arg, "--limit-storage")) {
-            if (command != .start) fatal("--limit-storage: supported only by 'start' command", .{});
-            storage_size_limit = parse_flag("--limit-storage", arg);
-        } else if (mem.startsWith(u8, arg, "--cache-grid")) {
-            if (command != .start) fatal("--cache-grid: supported only by 'start' command", .{});
-            cache_grid = parse_flag("--cache-grid", arg);
-        } else if (mem.eql(u8, arg, "--verbose")) {
-            if (command != .version) fatal("--verbose: supported only by 'version' command", .{});
-            verbose = true;
-        } else if (mem.eql(u8, arg, "-h") or mem.eql(u8, arg, "--help")) {
-            std.io.getStdOut().writeAll(usage) catch os.exit(1);
-            os.exit(0);
-        } else if (mem.startsWith(u8, arg, "-")) {
-            fatal("unexpected argument: '{s}'", .{arg});
-        } else if (path == null) {
-            if (!(command == .format or command == .start)) fatal("unexpected path", .{});
-            path = arg;
-        } else {
-            fatal("unexpected argument: '{s}' (must start with '--')", .{arg});
-        }
-    }
-
-    switch (command) {
-        .version => {
+    switch (cli_args) {
+        .version => |version| {
             return Command{
                 .version = .{
                     .args_allocated = args,
-                    .verbose = verbose orelse false,
+                    .verbose = version.verbose,
                 },
             };
         },
-        .format => {
-            const replica_count_parsed = parse_replica_count(
-                replica_count orelse fatal("required: --replica-count", .{}),
```

Comment:
I tried to make all error message consistent, and to pub the problematic flag name first.